src/serializers/resource.js

```diff
@@ -73,38 +97,31 @@ function ResourceSerializer(Implementation, model, records, integrator, opts, me
               return;
             }
 
-            var fieldReference = referenceSchema.idField;
-
-            if (_.isArray(field.type) && !fieldReference && referenceSchema.isVirtual) {
-              if (_.find(referenceSchema.fields, function (field) { return field.field === 'id'; })) {
-                fieldReference = 'id';
-              } else {
-                logger.warn('Cannot find the \'idField\' attribute in your \'' +
-                  referenceSchema.name + '\' Smart Collection declaration.');
-              }
-            }
+            const fieldReference = getReferenceField(referenceSchema, field);
 
             _.each(referenceSchema.fields, function (field) {
               detectFieldWithSpecialFormat(field, fieldName);
             });
 
+            const modelName = Implementation.getModelName(model);
+            const relatedFunction = field.relationship === 'BelongsTo'
+              ? null
+              : getRelatedLinkForHasMany(modelName, field.field);
             dest[fieldName] = {
               ref: fieldReference,
+              nullIfMissing: true,
```

Comment:
We need this or the links will not be included if the relation is not in the records sent to the serializer